### ISSUE TITLE
New-IntuneWin32AppDetectionRuleScript.ps1 Corrupted Detection Scripts on converting to Base64

### ISSUE THREAD

**Issue by joshinryz** [MSEndpointMgr/IntuneWin32App#15]
Using non-utf encoding of multi-character line scripts breaks when imported into Intune.

I change line 49 from:
$ScriptContent = [System.Convert]::ToBase64String([System.IO.File]::ReadAllBytes("$($ScriptFile)"))

to:

$ScriptContent = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($(get-content $ScriptFile) -join [Environment]::NewLine))

To resolve this. I'll submit a PR later after some more tesitng.

**Comment by NickolajA** [comment]
Fixed this in 1.3.1 that will be released shortly. Please verify and report back :)

**Comment by b-sarrazin** [comment]
Hello,

My detection script is in UTF-8.
I had to re-edit the file to replace line 49 with:
`$ScriptContent = [System.Convert]::ToBase64String([System.IO.File]::ReadAllBytes("$($ScriptFile)"))`

Without this modification, the detection file always came out successfully and did not install the app since it was detected as already installed ...

If you need more info, don't hesitate.

Thank you for this fantastic module!
I deploy around fifty applications. It saves me a lot of time.

**Comment by dbalcomb** [comment]
I came across this issue today. It appears that the change introduced in 1.3.1 is not working as intended because my detection status is always successful and the intune management logs show that the output is a list of bytes. I have tested both lines in the original comment and both produce the correct output.

**Comment by denismcapple** [comment]
Thank you @[b-sarrazin] - I was pulling my hair out for a while on this and came across your post!

Thanks for this awesome module :)

**Comment by Ekrist3** [comment]
Thanks everyone. I have had the same issue as the rest and @[b-sarrazin] did make the correct remediation.
I think [joshinryz] in the initial post most have had some encoding problems.
Thanks for the great module.

**Comment by jason-nyc** [comment]
This module is awesome.  But the UTF8 encoding bug tripped me up for days since it's hard to debug this type of issue.  What's worse, it seems like it was working fine in earlier versions, but then @joshinryz asked @NickolajA to change it for some reason.

The error is in these two files:
New-IntuneWin32AppRequirementRuleScript.ps1
New-IntuneWin32AppDetectionRuleScript.ps1

The issue:
It has to do with the way the local .ps1 file is parsed and published to Intune (which expects Base64 encoding, not UTF8 encoding, or something along those lines).  After looking at the logs I noticed that, instead of StdOut from the ps1, the log was showing the ascii codes of the .ps1 file itself.  The editors I use (Notepad++, Powershell ISE, Notepad) all work fine after this fix.

The fix:
Replace this
$ScriptContent = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes([System.IO.File]::ReadAllBytes("$($ScriptFile)") -join [Environment]::NewLine))

With this
$ScriptContent = [System.Convert]::ToBase64String([System.IO.File]::ReadAllBytes("$($ScriptFile)"))

**Comment by Skeb1ns** [comment]
I'm glad that I stumbled upon this issue. I thought it was me and couldn't figure out why Intune always reported that the app was installed even though I expected it to continue with the installation. Would be great if the functions get an update.

**Comment by NickolajA** [comment]
Yeah I'll update this again, apparently didn't think it through fully.

**Comment by Skeb1ns** [comment]
That being said, your module is awesome and saves us a lot of time!

**Comment by NickolajA** [comment]
`$ScriptContent = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes((Get-Content -Path "$($ScriptFile)" -Raw -Encoding UTF8)))`

Tested this and it seem to work, would someone be able to validate?

**Comment by b-sarrazin** [comment]
> `$ScriptContent = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes((Get-Content -Path "$($ScriptFile)" -Raw -Encoding UTF8)))`
> 
> Tested this and it seem to work, would someone be able to validate?

It also works on my side
